### PR TITLE
Feature color and size selection in product page and reflected to add to cart page.

### PR DESCRIPTION
**Pull request by katepallit**

### CODE REVIEW

File: src/views/Product.vue [hunk 1]
```diff
@@ -283,5 +308,7 @@ export default {
     letter-spacing: 0.5px;
     text-transform: uppercase;
   }
-
+  .error {
+    color: red;
```
Comment (dinesh-nagrani) [inline src/views/Product.vue:312]: use color variable

Comment (katepallit) [reply src/views/Product.vue:335]: ok

File: src/components/product/ProductSize.vue [hunk 2]
```diff
@@ -1,75 +1,97 @@
...
+		width: 30px;
+		height: 30px;
+		display: flex;
+		cursor: pointer;
+		font-size: 0.8rem;
+		margin-right: 1rem;
+		border-radius: 50%;
+		align-items: center;
+		justify-content: center;
+		border: 2px solid #ccc;
```
Comment (dinesh-nagrani) [inline src/components/product/ProductSize.vue:77]: Please take latest pull to have color variables and add color variables if you have added them.

File: src/components/plp/ProductTile.vue [hunk 3]
```diff
@@ -29,10 +29,11 @@ export default {
...
     },
     methods: {
-        onClick() {
-            this.$store.dispatch('product/setProduct', this.product)
-            this.$store.dispatch("category/getSpeceficProduct", this.product.sku);
+        async onClick() {
+            await this.$store.dispatch('product/setProduct', this.product)
+            await this.$store.dispatch("category/getSpeceficProduct", this.product.sku);
             this.$router.push({ path: `/category/men/${this.product.sku}` })
+            this.$store.dispatch("product/setSelectedSize", 0);
```
Comment (dinesh-nagrani) [inline src/components/plp/ProductTile.vue:36]: Instead calling store here, as we already give a call in first line to store to set product , can we handle the logic there of size.

File: src/components/utilityNav/MiniCart.vue [hunk 4]
```diff
@@ -1,131 +1,77 @@
...
-                    <span>${{grossPrice}}</span>
-                </div>
-                <div class="mini-cart-button">
-                    <button class="btn btn-primary" @click="viewCart">View Cart</button>
-                    <button class="btn btn-primary" @click="checkout">Checkout</button>
-                </div>
-            </div>
-        </template>
-    </li>
+		<span class="count > 0">{{itemCount}}</span>
```
Comment (wasil-sayyed) [inline src/components/utilityNav/MiniCart.vue:6]: File not formatted.

File: src/components/plp/ProductTile.vue (hunk 3 above)
Comment (katepallit) [reply src/components/plp/ProductTile.vue:36]: We are using this to reset the size when the product changes.

**Pull request merged**